src/core/src/package_managers/AptitudePackageManager.py

```diff
@@ -359,6 +367,73 @@ def get_package_size(self, output):
+        try:
+            self.composite_logger.log_debug("Disabling auto OS updates if they are enabled")
+            self.backup_image_default_patch_mode()
+            self.update_image_default_patch_mode(self.update_package_list, "0")
+            self.update_image_default_patch_mode(self.unattended_upgrade, "0")
+            self.composite_logger.log("Successfully disabled auto OS updates")
+        except Exception as error:
+            self.composite_logger.log_error("Could not disable auto OS updates. [Error={0}]".format(repr(error)))
+
+    def backup_image_default_patch_mode(self):
+        """ Records the default system settings for auto OS updates within patch extension artifacts for future reference.
+        We only log the default patch_mode a VM comes with, any subsequent updates will not be recorded"""
+        try:
+            if not self.image_default_patch_mode_backup_exists():
+                image_default_patch_mode = self.env_layer.file_system.read_with_retry(self.image_default_patch_mode_file_path)
+                settings = image_default_patch_mode.strip().split('\n')
+                for patch_mode in settings:
+                    if self.update_package_list in str(patch_mode):
+                        self.update_package_list_value = re.search(self.update_package_list + ' *"(.*?)".', str(patch_mode)).group(1)
+                    if self.unattended_upgrade in str(patch_mode):
+                        self.unattended_upgrade_value = re.search(self.unattended_upgrade + ' *"(.*?)".', str(patch_mode)).group(1)
+
+                backup_image_default_patch_mode_json = {
+                    self.update_package_list: self.update_package_list_value,
+                    self.unattended_upgrade: self.unattended_upgrade_value
+                }
+                self.composite_logger.log_debug("Logging default system settings for auto OS updates. [Settings={0}] [Log file path= {1}]"
+                                                .format(str(backup_image_default_patch_mode_json), self.image_default_patch_mode_backup_path))
+                self.env_layer.file_system.write_with_retry(self.image_default_patch_mode_backup_path, '{0}'.format(json.dumps(backup_image_default_patch_mode_json)), mode='w+')
+        except Exception as error:
+            error_message = "Exception during fetching and logging default auto update settings on the machine. [Exception={0}]".format(repr(error))
+            self.composite_logger.log_error(error_message)
+            raise
+
+    def is_image_default_patch_mode_backup_valid(self, image_default_patch_mode_backup):
+        if self.update_package_list in image_default_patch_mode_backup or self.unattended_upgrade in image_default_patch_mode_backup:
+            return True
+        return False
+
+    def update_image_default_patch_mode(self, patch_mode, value="0"):
```

Comment:
update_os_patch_mode_settings

Reply:
The confusion with the name as it it now is it made me wonder if you were overwriting the backed up image_default setting. image_default = original settings; not current settings

Reply:
Also the first parameter is not really patch_mode. Patch_mode is only if it is restricted to one of the settings we have as patch_mode across all our components. Please check all variable and function names for clarity - in this case, I'd use something like os_patch_configuration_sub_setting (or something similar).

Reply:
One question, if the file the settings are being read from is referred to as os_patch_mode_settings_file, then should the individual settings within it be, os_patch_mode_setting or os_patch_mode_sub_setting? This function updates only one setting every invocation in the os_patch_mode_settings_file, so having the function name as update_os_patch_mode_settings and the param name as os_patch_configuration_sub_setting seems like a mismatch.

Reply:
Using os_patch_mode_settings and os_patch_mode_sub_setting for now, let me know if that makes sense

Reply:
Using patch_configuration instead of patch_mode for everything on the VM or extension, like discussed. patch_mode to be used for any setting that comes into the extension